packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -50,6 +50,16 @@ export interface PluginPackage {
     contributes?: PluginPackageContribution;
     packagePath: string;
 }
+export namespace PluginPackage {
+    /**
+     * @param relativePath should be normalized, i.e. should not contain `.` or `..`,
+     * otherwise they will be removed by a browser leading to a bogus path,
+     * see https://stackoverflow.com/questions/3856693/a-url-resource-that-is-a-dot-2e
+     */
+    export function toPluginUrl(pck: PluginPackage, relativePath: string): string {
+        return `hostedPlugin/${getPluginId(pck)}/${relativePath}`;
```

Comment:
If this is an URL, shouldn't we encode the segments?
```ts
return encodeURI(`hostedPlugin/${getPluginId(pck)}/${relativePath}`);
```

Or encore the segments with `encodeURIComponent`?

Perhaps it is easy to verify with a path with a special character or with whitespace.

Reply:
+1 encoding will make sense if paths contains spaces for sure

Reply:
I've added encoding of a part part as a component that it excludes issues with `/../` in URl